### PR TITLE
Add new configs for full-text indexes

### PR DESCRIPTION
**Pull request by NataliaIvakina**
based on the PR https://github.com/neo-technology/neo4j/pull/25745

See also https://github.com/neo-technology/neo4j/pull/25885

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1672.surge.sh

Updated pages:
- [Configuration settings](https://neo4j-docs-operations-1672.surge.sh/operations-manual/5/configuration/configuration-settings)

File: modules/ROOT/pages/configuration/configuration-settings.adoc [hunk 1]
```diff
@@ -2375,6 +2375,52 @@ m|+++false+++
 |===
 
 
+[role=label-new--5.21]
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2378]:
```suggestion
[role=label--new-5.21]
```

File: modules/ROOT/pages/configuration/configuration-settings.adoc [hunk 2]
```diff
@@ -2375,6 +2375,52 @@ m|+++false+++
...
+|===
+|Description
+a|The number of threads that are applying the queued index updates for eventually consistent full-text indexes.
+|Valid values
+a|An integer
+|Default value
+m|+++1+++
+|===
+
+[role=label-new--5.21]
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2393]:
```suggestion
[role=label--new-5.21]
```

File: modules/ROOT/pages/configuration/configuration-settings.adoc [hunk 3]
```diff
@@ -2375,6 +2375,52 @@ m|+++false+++
...
+|===
+|Description
+a|How often an eventually consistent full-text index will be refreshed (readers guaranteed to see changes). If set to `0` then refresh is done by the threads applying eventually consistent fulltext index updates.
+|Valid values
+a|Duration
+|Default value
+m|+++0s+++
+|===
+
+[role=label-new--5.21]
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2408]:
```suggestion
[role=label--new-5.21]
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1672.surge.sh

Updated pages:
- [Configuration settings](https://neo4j-docs-operations-1672.surge.sh/operations-manual/5/configuration/configuration-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1672.surge.sh

Updated pages:
- [Configuration settings](https://neo4j-docs-operations-1672.surge.sh/operations-manual/5/configuration/configuration-settings)

File: modules/ROOT/pages/configuration/configuration-settings.adoc [hunk 4]
```diff
@@ -2375,6 +2375,52 @@ m|+++false+++
...
 
+[role=label--new-5.21]
+[[config_db.index.fulltext.eventually_consistent_apply_parallelism]]
+=== `db.index.fulltext.eventually_consistent_apply_parallelism`
+
+.db.index.fulltext.eventually_consistent_apply_parallelism
+[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
+|===
+|Description
+a|The number of threads that are applying the queued index updates for eventually consistent full-text indexes.
```
Comment (renetapopova) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2386]: This also needs to be changed and merged in the codebase before the cut-off on Friday.

File: modules/ROOT/pages/configuration/configuration-settings.adoc [hunk 5]
```diff
@@ -2375,6 +2375,52 @@ m|+++false+++
...
+
+[role=label--new-5.21]
+[[config_db.index.fulltext.eventually_consistent_refresh_interval]]
+=== `db.index.fulltext.eventually_consistent_refresh_interval`
+
+.db.index.fulltext.eventually_consistent_refresh_interval
+[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
+|===
+|Description
+a|How often an eventually consistent full-text index will be refreshed (readers guaranteed to see changes). If set to `0` then refresh is done by the threads applying eventually consistent fulltext index updates.
```
Comment (renetapopova) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2401]:
```suggestion
a|How often an eventually consistent full-text index is refreshed (readers are guaranteed to see changes). If set to `0`, refresh is done by the threads applying eventually consistent full-text index updates.
```

(readers are guaranteed to see changes) sounds strange to me for a config description. Who are the readers?

Comment (renetapopova) [reply modules/ROOT/pages/configuration/configuration-settings.adoc:2401]: Same here - needs to be merged in the codebase before the cut-off.

File: modules/ROOT/pages/configuration/configuration-settings.adoc [hunk 6]
```diff
@@ -2375,6 +2375,52 @@ m|+++false+++
...
+
+[role=label--new-5.21]
+[[config_db.index.fulltext.eventually_consistent_refresh_parallelism]]
+=== `db.index.fulltext.eventually_consistent_refresh_parallelism`
+
+.db.index.fulltext.eventually_consistent_refresh_parallelism
+[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
+|===
+|Description
+a|The number of threads that can do full-text index refresh in parallel, basically how many eventually consistent full-text indexes can be refreshed in parallel.
```
Comment (renetapopova) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2416]:
```suggestion
a|The number of threads that can do full-text index refresh in parallel, i.e. the number of eventually consistent full-text indexes that can be refreshed in parallel.
```

Comment (renetapopova) [reply modules/ROOT/pages/configuration/configuration-settings.adoc:2416]: Perhaps we can just use the second sentence.

Same here - needs to be merged in the codebase before the cut-off.

**Review by renetapopova — CHANGES_REQUESTED** [review]
Thanks, @NataliaIvakina. I added some comments to the descriptions. Just make sure the descriptions align with the ones in the codebase.

File: modules/ROOT/pages/configuration/configuration-settings.adoc (hunk 4 above)
Comment (NataliaIvakina) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2386]:
```suggestion
a|The number of threads that are applying to the queued index updates for eventually consistent full-text indexes.
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1672.surge.sh

Updated pages:
- [Configuration settings](https://neo4j-docs-operations-1672.surge.sh/operations-manual/5/configuration/configuration-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1672.surge.sh

Updated pages:
- [Configuration settings](https://neo4j-docs-operations-1672.surge.sh/operations-manual/5/configuration/configuration-settings)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1672.surge.sh

Updated pages:
- [Configuration settings](https://neo4j-docs-operations-1672.surge.sh/operations-manual/5/configuration/configuration-settings)

File: modules/ROOT/pages/configuration/configuration-settings.adoc [hunk 7]
```diff
@@ -2375,6 +2375,52 @@ m|+++false+++
...
+
+[role=label--new-5.21]
+[[config_db.index.fulltext.eventually_consistent_refresh_interval]]
+=== `db.index.fulltext.eventually_consistent_refresh_interval`
+
+.db.index.fulltext.eventually_consistent_refresh_interval
+[frame="topbot", stripes=odd, grid="cols", cols="<1s,<4"]
+|===
+|Description
+a|How often an eventually consistent full-text index is refreshed (readers are guaranteed to see changes). If set to `0`, refresh is done by the threads applying eventually consistent full-text index updates.
```
Comment (NataliaIvakina) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2401]: readers or secondaries?

**Review by renetapopova — APPROVED** [review]
If it is aligned with codebase, I approve it. Otherwise, we need to keep the same descriptions as the ones there.

File: modules/ROOT/pages/configuration/configuration-settings.adoc (hunk 7 above)
Comment (NataliaIvakina) [inline modules/ROOT/pages/configuration/configuration-settings.adoc:2401]:
```suggestion
a|How often an eventually consistent full-text index is refreshed (changes are guaranteed to be visible). If set to `0`, refresh is done by the threads applying eventually consistent full-text index updates.
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1672.surge.sh

Updated pages:
- [Configuration settings](https://neo4j-docs-operations-1672.surge.sh/operations-manual/5/configuration/configuration-settings)

**Pull request merged**